src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -114,42 +166,93 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
+
+                        // Check whether scope is set to inner or outer
+                        var scope = currentProcessedResource.InsensitiveToken("properties.expressionEvaluationOptions")?.InsensitiveToken("scope")?.ToString();
+                        //string nextPathPrefix = $".properties.template.";
+                        string nextPathPrefix = nestedTemplate.Path;
+                       
+                        IEnumerable<IEvaluation> result;
+
+                        if (scope == null || scope == "outer")
+                        {
+                            // Variables, parameters and functions inherited from parent template
+                            string functionsKey = populatedNestedTemplate.InsensitiveToken("functions")?.Parent.Path;
+                            string variablesKey = populatedNestedTemplate.InsensitiveToken("variables")?.Parent.Path;
+                            string parametersKey = populatedNestedTemplate.InsensitiveToken("parameters")?.Parent.Path;
+
+                            if (functionsKey != null)
+                            {
+                                populatedNestedTemplate[functionsKey] = jsonTemplate.InsensitiveToken("functions");
+
+                            }
+                            if (variablesKey != null)
+                            {
+                                populatedNestedTemplate[variablesKey] = jsonTemplate.InsensitiveToken("variables");
+
+                            }
+                            if (parametersKey != null)
+                            {
+                                populatedNestedTemplate[parametersKey] = jsonTemplate.InsensitiveToken("parameters");
+                            }
+                        }
+                        else // scope is inner
+                        {
+                            // Pass variables and functions to child template
+                            (populatedNestedTemplate.InsensitiveToken("variables") as JObject)?.Merge(currentProcessedResource.InsensitiveToken("properties.variables"));
+                            (populatedNestedTemplate.InsensitiveToken("functions") as JObject)?.Merge(currentProcessedResource.InsensitiveToken("properties.functions)"));
+
+                            // Pass parameters parameters to child template as the 'parameters' argument
+                            var parametersToPass = currentProcessedResource.InsensitiveToken("properties.parameters");
+
+                            if (parametersToPass != null)
+                            {
+                                parametersToPass["parameters"] = parametersToPass;
+                                parametersToPass["$schema"] = "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#";
+                                parametersToPass["contentVersion"] = "1.0.0.0";
+                                parameters = JsonConvert.SerializeObject(parametersToPass);
```

Comment:
I'm not confident you actually need to do all this.  Try just passing the parameters as a string as they are, without adding all the metadata fields.

Reply:
I did and it had an error requiring those fields, which is why I added them.

Reply:
Ok, thanks.